### PR TITLE
vesktop: init

### LINKED ISSUE #364: Vesktop (Vencord) theme
Vesktop is an alternative client for Discord which bundles Vencord. Vencord allows for BetterDiscord CSS themes to be loaded.

### PR DESCRIPTION
**Pull request by itsvic-dev**
Adds a generated theme to Vesktop. (might need extra tweaks for the colors)

image

image

Closes #364

### CODE REVIEW

**Comment by itsvic-dev** [comment]
image

File: modules/vesktop/hm.nix [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{ config, lib, ... }:
+let
+  themeFile = config.lib.stylix.colors {
+    template = ./template.mustache;
+    extension = ".css";
+  };
+in
+{
+  options.stylix.targets.vesktop.enable = config.lib.stylix.mkEnableTarget "Vesktop" true;
+  config = lib.mkIf config.stylix.targets.vesktop.enable {
```
Comment (trueNAHO) [inline modules/vesktop/hm.nix:10]:
```suggestion

  config = lib.mkIf config.stylix.targets.vesktop.enable {
```

File: modules/vesktop/template.mustache [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    --interactive-hover: var(--base05);
+    --interactive-active: var(--base07);
+    --interactive-muted: var(--base03);
+    --channels-default: var(--base04);
+    --channel-icon: var(--base04);
+    --header-primary: var(--base06);
+    --header-secondary: var(--base03);
+    --scrollbar-thin-track: transparent;
+    --scrollbar-auto-track: transparent;
+}
```
Comment (trueNAHO) [inline modules/vesktop/template.mustache:57]: No newline at EOF.

**Review by trueNAHO — APPROVED** [review]
LGTM, except for two nitpicks.

**Comment by itsvic-dev** [comment]
applied the suggestions and rebased to master 👍🏻

**Review by trueNAHO — APPROVED** [review]

**Comment by danth** [comment]
There has previously been some discussion on whether we want to support modifications which may break the terms of service of Discord. However, it appears that this module only takes effect if the user has chosen to install Vesktop, in which case they should already be aware they're using a modified client.

In terms of the theme itself, the color for unread channels looks a bit unusual, but that may just be your particular config. Are you using a `base16` scheme or one generated automatically by Stylix?

**Comment by itsvic-dev** [comment]
I'm using the color scheme generated by Stylix.
I agree that it is a bit unusual, so some experimentation with the theme might be needed. But I think that this could be addressed in a separate PR.


On May 15, 2024 10:06, Daniel Thwaites ***@***.***> wrote:

There has previously been some discussion on whether we want to support modifications which may break the terms of service of Discord. However, it appears that this module only takes effect if the user has chosen to install Vesktop, in which case they should already be aware they're using a modified client.

In terms of the theme itself, the color for unread channels looks a bit unusual, but that may just be your particular config. Are you using a base16 scheme or one generated automatically by Stylix?

**Comment by danth** [comment]
Yeah okay, that's an issue with the theme generator and not this module.

**Review by danth — APPROVED** [review]
Looks good to me.

Is this ready to merge?

**Comment by itsvic-dev** [comment]
I believe so, yes.

On May 20, 2024 12:11, Daniel Thwaites ***@***.***> wrote:

@danth approved this pull request.

Looks good to me.

Is this ready to merge?

**Pull request merged**